tool/dartpad_picker/web/dartpad_picker_main.dart

```diff
@@ -13,224 +15,41 @@ void main() {
 
   var dartPadHost = querySelector('#dartpad-host');
   var select = querySelector('#dartpad-select');
-
-  var snippets = [
-    Snippet('Hello world', helloWorld),
-    Snippet('Functions', functions),
-    Snippet('Control flow', controlFlow),
-    Snippet('Strings', strings),
-    Snippet('Collection literals', collectionLiterals),
-    Snippet('Classes', classes),
-    Snippet('Compute Pi', piMonteCarlo),
+  final examples = querySelector('#try-dart-examples');
+
+  final exampleNames = [
+    'Hello world',
+    'Functions',
+    'Control flow',
+    'Strings',
+    'Collection literals',
+    'Classes',
+    'Compute Pi',
   ];
 
-  DartPadPicker(dartPadHost, select, snippets);
+  final snippets = exampleNames.map((n) => _createSnippet(examples, n));
+
+  DartPadPicker(dartPadHost, select, snippets.toList());
 }
 
+Snippet _createSnippet(HtmlElement root, String name) => Snippet(
+      name,
+      _getSrc(
+        root,
+        '#try-dart-${name.toLowerCase().replaceAll(' ', '-')}',
+      ),
+    );
+
+String /*?*/ _getSrc(HtmlElement root, String id, [String fallback]) =>
```

Comment:
What does the `/*?*/` do?

Reply:
Oh, thanks for catching that. The function originally returned a nullable String, but it doesn't anymore. (I use `/*?*/` to mark types as nullable, until we can use the `?` syntax for real :)).

Reply:
Removed comment.